transformer/timestamp.go

```diff
@@ -70,16 +71,22 @@ func (config *Timestamp) Transform(c *skogul.Container) error {
 
 		obj, err := skogul.ExtractNestedObject(metric.Data, config.Source)
 		if err != nil {
-			timestampLogger.Warning("Unable to extract timestamp field from a metric")
-			return skogul.Error{Reason: "Failed to extract timestamp field from a metric"}
+			return fmt.Errorf("failed to extract timestamp field from a metric: %w", err)
 		}
 		timestamp, ok := obj[config.Source[len(config.Source)-1]].(string)
 
 		if !ok {
+			// XXX: I imagine this could easily be a log-bomb.
+			// Why log it as ERROR if we don't care enough to
+			// propagate the error up?
 			timestampLogger.Error("Failed to cast timestamp field to a string")
 			if config.Fail {
-				return skogul.Error{Reason: "Failed to cast timestamp field to a string"}
+				return fmt.Errorf("failed to cast timestamp field to a string")
 			}
+			// XXX: Added late, is there a use case where
+			// timestamp isn't a string but we still want to
+			// parse it? O_o
```

Comment:
That's fair enough, but the transformer wouldn't be able to actually parse it as a timestamp, so it makes little sense to try? time.Parse() expects a string, so if the casting fails, it will fail as well.

Reply:
Yes, but we could use `time.Unix()` instead for "parsing" this - but we'd need to know if it's in seconds, milliseconds (javascript...), nanoseconds, ... Currently, we don't support this at all.